src/fu-engine-repair.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2015 Kate Hsuan <hpa@redhat.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN "FuEngine"
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <glib/gi18n.h>
+#include <glib/gstdio.h>
+#ifdef HAVE_GIO_UNIX
+#include <gio/gunixfdlist.h>
+#include <glib-unix.h>
+#endif
+#include <locale.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "fu-console.h"
+#include "fu-engine.h"
+#include "fu-plugin-private.h"
+#include "fu-engine-repair.h"
+#include "fu-util-common.h"
+
+
+static gboolean
+grubby_set(gboolean enable, const gchar *grubby_arg,  GError **error)
+{
+	g_autofree gchar *output = NULL;
+	g_autofree gchar *arg_string = NULL;
+	const gchar *argv_grubby[] = {"",
+				      "--update-kernel=DEFAULT",
+				      "",
+				      NULL};
+	g_autofree gchar *grubby = NULL;
+	grubby = fu_path_find_program("grubby", NULL);
+	argv_grubby[0] = grubby;
```

Comment:
not all distros support grubby.  I think you need to check that you got a successful result and return an error otherwise.

Reply:
Specifically I think you should check early on whether `grubby` is supported to decide whether or not to offer changing kernel command line options in this fashion.

Reply:
Sure, I will implement this check. :)